crates/da-clients/celestia/src/lib.rs

```diff
@@ -0,0 +1,190 @@
+
+        // External Id : Height of Block the blob is submitted to.
+        // 2 ways to check :
+        // Is the current block ahead of the block that we have, if it is does my block contain the blob.
+        // Call the blob_get_all function of the client
+
+        let height_id = external_id.parse()?;
+        let retrieved_blobs = self.celestia_client.blob_get_all(height_id, &[self.nid]).await;
+
+        //TODO: Assumption: Given that we are sending only 1 nid, we'll get an array of 1 object back.
+
+        match retrieved_blobs {
+            Ok(blobs) => {
+                if blobs.len() == 1 {
+                    Ok(DaVerificationStatus::Verified)
+                } else {
+                    Ok(DaVerificationStatus::Rejected(format!("Expected 1 blob, but got {}", blobs.len())))
+                }
+            }
+            Err(e) => Ok(DaVerificationStatus::Rejected(format!("Error occurred: {}", e))),
+        }
+    }
+
+    async fn max_blob_per_txn(&self) -> u64 {
+        //Info: No docs suggest a number, default to 1.
+        1
+    }
+
+    async fn max_bytes_per_blob(&self) -> u64 {
+        //Info: https://github.com/celestiaorg/celestia-node/issues/3356
+        1974272
+    }
+}
+
+impl TryFrom<config::CelestiaDaConfig> for CelestiaDaClient {
+    type Error = anyhow::Error;
+    fn try_from(conf: config::CelestiaDaConfig) -> Result<Self, Self::Error> {
+        // Borrowed the below code from https://github.com/eigerco/lumina/blob/ccc5b9bfeac632cccd32d35ecb7b7d51d71fbb87/rpc/src/client.rs#L41.
+        // Directly calling the function wasn't possible as the function is async. Since
+        // we only need to initiate the http provider and not the ws provider, we don't need async
+
+        let mut headers = HeaderMap::new();
+
+        // checking if Auth is available
+        if let Some(auth_token) = conf.auth_token {
+            let val = HeaderValue::from_str(&format!("Bearer {}", auth_token))?;
+            headers.insert(header::AUTHORIZATION, val);
+        }
+
+        let http_client = HttpClientBuilder::default()
+            .set_headers(headers)
+            .build(conf.http_provider.as_str())
+            .map_err(|e| CelestiaDaError::Client(format!("could not init http client: {e}")))?;
+
+        // Convert the input string to bytes
+        let bytes = conf.nid.as_bytes();
+
+        // Create a new Namespace from these bytes
+        let nid = Namespace::new_v0(bytes)
+            .map_err(|e| CelestiaDaError::Generic(format!("could not init namespace: {e}")))
+            .unwrap();
+
+        Ok(Self { celestia_client: http_client, nid })
+    }
+}
+
+#[cfg(test)]
+mod tests {
+
+    use super::*;
+    use config::{CelestiaDaConfig, DEFAULT_CELESTIA_NODE, DEFAULT_NID};
+
+    #[tokio::test]
+    #[ignore = "Can't run without manual intervention, setup celestia-node and fund address."]
```

Comment:
let's add the steps to run these test cases over here as a comment

Reply:
Noted